src/toil/resource.py

```diff
@@ -557,35 +578,35 @@ def _resourcePath(self):
             return self.dirPath
 
     @classmethod
-    def _initModuleName(cls, dirPath):
+    def _initModuleName(cls, dirPath: str) -> Optional[str]:
         for name in ('__init__.py', '__init__.pyc', '__init__.pyo'):
             if os.path.exists(os.path.join(dirPath, name)):
                 return name
         return None
 
-    def _rootPackage(self):
+    def _rootPackage(self) -> Any:
         try:
             head, tail = self.name.split('.', 1)
         except ValueError:
-            raise ValueError('%r is stand-alone module.' % self)
+            raise ValueError('%r is stand-alone module.' % self.__repr__())
         else:
             return head
 
-    def toCommand(self):
+    def toCommand(self) -> Tuple[str, ...]:
```

Comment:
Maybe use `Sequence[str]` as the return type so you can say all the elements are `str` and not just the first one?

Reply:
Then it would be allowable to pass the result of `toCommand()` to `fromCommand()`, which would make sense.